cirq/study/resolver.py

```diff
@@ -89,18 +89,33 @@ def value_of(self,
         Returns:
             The value of the parameter as resolved by this resolver.
         """
-        # Input is a float, no resolution needed: return early
-        if isinstance(value, float):
-            return value
+
+        def pass_through(val) -> Optional[Any]:
```

Comment:
What are your thoughts about making this a private function (e.g. _sympy_pass_through) with a short docstring rather than an inline nested function?  I personally dislike nested functions, though I don't know if this is a global sentiment.

Google style guide at least says "Avoid nested functions or classes except when closing over a local value."

Reply:
I'm open to it - the method is not used anywhere else that's why I kept it nested but I'm happy to move it out.